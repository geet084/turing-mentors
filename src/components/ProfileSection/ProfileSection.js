import React from 'react';
import Collapsible from 'react-collapsible';

export const ProfileSection = (props) => {
  const { content, name } = props;

  return (
    <Collapsible
      openedClassName="profile"
      contentInnerClassName="profile"
      contentOuterClassName="profile"
      trigger={`${name}  ▼`}
      triggerWhenOpen={`${name}  ▲`}
    >
      {content}
    </Collapsible>
  )
}

export default ProfileSection;