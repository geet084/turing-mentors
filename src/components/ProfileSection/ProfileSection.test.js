import React from 'react';
import { shallow } from 'enzyme';
import { ProfileSection } from './ProfileSection';

describe('ProfileSection', () => {
  let wrapper;

  it('should match the correct snapshot', () => {
    wrapper = shallow(<ProfileSection />);
    expect(wrapper).toMatchSnapshot();
  });
});