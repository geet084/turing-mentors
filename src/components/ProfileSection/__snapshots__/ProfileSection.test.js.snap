// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSection should match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileSection />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": undefined,
      "className": "",
      "classParentString": "Collapsible",
      "contentContainerTagName": "div",
      "contentInnerClassName": "profile",
      "contentOuterClassName": "profile",
      "easing": "linear",
      "lazyRender": false,
      "onClose": [Function],
      "onClosing": [Function],
      "onOpen": [Function],
      "onOpening": [Function],
      "open": false,
      "openedClassName": "profile",
      "overflowWhenOpen": "hidden",
      "tabIndex": null,
      "transitionCloseTime": null,
      "transitionTime": 400,
      "trigger": "undefined  ▼",
      "triggerClassName": "",
      "triggerDisabled": false,
      "triggerOpenedClassName": "",
      "triggerSibling": null,
      "triggerStyle": null,
      "triggerTagName": "span",
      "triggerWhenOpen": "undefined  ▲",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "className": "",
        "classParentString": "Collapsible",
        "contentContainerTagName": "div",
        "contentInnerClassName": "profile",
        "contentOuterClassName": "profile",
        "easing": "linear",
        "lazyRender": false,
        "onClose": [Function],
        "onClosing": [Function],
        "onOpen": [Function],
        "onOpening": [Function],
        "open": false,
        "openedClassName": "profile",
        "overflowWhenOpen": "hidden",
        "tabIndex": null,
        "transitionCloseTime": null,
        "transitionTime": 400,
        "trigger": "undefined  ▼",
        "triggerClassName": "",
        "triggerDisabled": false,
        "triggerOpenedClassName": "",
        "triggerSibling": null,
        "triggerStyle": null,
        "triggerTagName": "span",
        "triggerWhenOpen": "undefined  ▲",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
